### PR TITLE
build: add path mapping support to broccoli typescript

### PR DESCRIPTION
**Pull request by hansl**

### CODE REVIEW

File: lib/broccoli/broccoli-typescript.js [hunk 1]
```diff
@@ -306,10 +382,41 @@ class CustomLanguageServiceHost {
       // so we we just return undefined when the path is not correct.
       return undefined;
     }
-
     return ts.ScriptSnapshot.fromString(fs.readFileSync(absoluteTsFilePath, FS_OPTS));
   }
 
+  resolveModuleNames(moduleNames, containingFile)/*: ResolvedModule[]*/ {
+    return moduleNames.map((moduleName) => {
+      for (const ext of ['ts', 'd.ts']) {
```
Comment (jeffbcross) [inline lib/broccoli/broccoli-typescript.js]: Why const here?

Comment (hansl) [inline lib/broccoli/broccoli-typescript.js]: It won't change.

Comment (jeffbcross) [inline lib/broccoli/broccoli-typescript.js]: Oh, I didn't realize `const` was block-scoped. Disregard.

**Comment by jeffbcross** [comment]
@hansl LGTM!

File: lib/broccoli/broccoli-typescript.js (hunk 1 above)
Comment (gkalpak) [inline lib/broccoli/broccoli-typescript.js]: Still why `const` here and `let` in other for-of loops ? :imp:

**Pull request merged**